### PR TITLE
Change tensor names order

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/frontends/onnx/frontend/src/input_model.cpp [hunk 1]
```diff
@@ -453,7 +453,8 @@ void InputModel::add_tensor_names(std::shared_ptr<Model>& model) {
         return std::find_if(std::begin(model_inputs),
                             std::end(model_inputs),
                             [&name](const OutputVector::value_type& input) {
-                                return input.get_names().count(name) > 0;
+                                return std::find(input.get_names().begin(), input.get_names().end(), name) !=
+                                       input.get_names().end();
```
Comment (ilya-lavrenov) [inline src/frontends/onnx/frontend/src/input_model.cpp:457]: should not we add some helpers like `ov::util::contains` (or copy https://github.com/openvinotoolkit/openvino/blob/master/src/inference/dev_api/ie_algorithm.hpp#L32-L35) to https://github.com/openvinotoolkit/openvino/blob/master/src/common/util/include/openvino/util/common_util.hpp ?

Comment (ilyachur) [reply src/frontends/onnx/frontend/src/input_model.cpp:457]: Maybe Tensor::has_name?

File: src/bindings/python/tests/test_frontend/test_frontend_onnx_editor.py [hunk 2]
```diff
@@ -1643,7 +1643,7 @@ def test_add_the_same_name_to_tensor_twice():
 
     add_input = ov_model.input(1)
     add_input_tensor_names = add_input.get_names()
-    assert len(add_input_tensor_names) == 2
+    assert len(add_input_tensor_names) == 3
```
Comment (t-jankowski) [inline src/bindings/python/tests/test_frontend/test_frontend_onnx_editor.py:1646]: Now the same string can be used several times as one of names of the same place. Is that purposely?

**Review by t-jankowski — CHANGES_REQUESTED** [review]

File: src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp [hunk 3]
```diff
@@ -330,8 +330,9 @@ void SetExeNetworkInfo(const std::shared_ptr<IExecutableNetworkInternal>& exeNet
         auto new_param = ov::as_type_ptr<ov::op::v0::Parameter>(param->copy_with_new_inputs({}));
         new_param->set_friendly_name(param_name);
         if (add_operation_names) {
+            const auto names = param->get_output_tensor(0).get_names();
```
[the remark this answers is not in the document]
Comment (ilyachur) [reply src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp]: Fixed

File: src/bindings/python/tests/test_frontend/test_frontend_onnx_editor.py (hunk 2 above)
Comment (ilyachur) [reply src/bindings/python/tests/test_frontend/test_frontend_onnx_editor.py:1646]: No, but before we prohibited this behavior by using `unordered_set`. This PR changes `unordered_set` to `list` in order to support order of tensor names in the future.

File: src/frontends/onnx/frontend/src/input_model.cpp (hunk 1 above)
Comment (ilya-lavrenov) [reply src/frontends/onnx/frontend/src/input_model.cpp:457]: ok, I agree

**Pull request closed without merging**